dkc/core/rest/file.py

```diff
@@ -88,9 +89,15 @@ def perform_create(self, serializer: FileSerializer):
             raise PermissionDenied()
         serializer.save(creator=user)
 
-    # TODO figure out how to indicate this response type in the OpenAPI schema
+    @swagger_auto_schema(
+        responses={
+            204: 'This file is pending or has no associated content.',
```

Comment:
I'm not sure if "no associated content" is an actual state that we support. I *think* that uploading a 0-byte file will still save an object to S3 and get downloaded via a redirect URL.

Reply:
Girder 3 supported 0 byte files but didn't represent them in the assetstore, they were always special cased. I am not sure whether that's good or bad from an implementation perspective, however I do think supporting them is a good idea since empty files are a valid concept and exist in the wild.

Reply:
Assuming that we properly support 0-byte files, a 204 will not be the response for one.

So, I'd just suggest rewording this
```suggestion
            204: 'This file is pending and has no associated content.',
```